pkg/scheduler/framework/plugins/nodeports/node_ports_test.go

```diff
@@ -144,12 +148,20 @@ func TestNodePorts(t *testing.T) {
 			p, _ := New(nil, nil)
 			cycleState := framework.NewCycleState()
 			_, preFilterStatus := p.(framework.PreFilterPlugin).PreFilter(context.Background(), cycleState, test.pod)
+			if test.wantPreFilterStatus != nil {
+				if diff := cmp.Diff(test.wantPreFilterStatus, preFilterStatus); diff != "" {
+					t.Errorf("preFilter: status does not match (-want,+got):\n%s", diff)
+				}
+				return
+			}
 			if !preFilterStatus.IsSuccess() {
```

Comment:
Seems weird to me. It means all `preFilterStatus` should be Success otherwise the test will fail, right?
What if the prefilter returns Skip then?

Reply:
guessing you should add `test.wantPreFilterStatus == nil &&` to this condition as my previous suggestion.
https://github.com/kubernetes/kubernetes/pull/115318#discussion_r1090136032

Reply:
Yeah, I think your suggestion is right, thanks.

Reply:
> Seems weird to me. It means all preFilterStatus should be Success otherwise the test will fail, right?

It's right and it should be. 
If pre-filter returns Skip (excepting Success) status and a test case has wantPreFilterStatus, it's compared in previous condition.
In other hand, a test case that has no wantPreFilterStatus should be success.
This allows us to determine if wantPreFilterStatus is needed.

Reply:
OK, I missed `return` in L158. So, you mean the test won't reach here if the PreFilter returns non-success including Skip. That makes sense.